flink-connector-oceanbase-cdc/src/main/java/com/ververica/cdc/connectors/oceanbase/source/OceanBaseRichParallelSourceFunction.java

```diff
@@ -0,0 +1,518 @@
+                snapshotConnection.close();
+            }
+        }
+
+        snapshotCompleted.set(true);
+    }
+
+    private void readSnapshotFromTable(Statement stmt, String databaseName, String tableName)
+            throws Exception {
+        // TODO make topic name configurable
+        String topicName = getDefaultTopicName(tenantName, databaseName, tableName);
+        Map<String, String> partition = getSourcePartition(tenantName, databaseName, tableName);
+        // the offset here is useless
+        Map<String, Object> offset = getSourceOffset(resolvedTimestamp);
+
+        String selectSql = String.format("SELECT * FROM `%s`.`%s`", databaseName, tableName);
+        ResultSet rs = stmt.executeQuery(selectSql);
+        ResultSetMetaData metaData = (ResultSetMetaData) rs.getMetaData();
+
+        String[] columnNames = new String[metaData.getColumnCount()];
+        int[] jdbcTypes = new int[metaData.getColumnCount()];
+        for (int i = 0; i < metaData.getColumnCount(); i++) {
+            columnNames[i] = metaData.getColumnName(i + 1);
+            jdbcTypes[i] =
+                    OceanBaseJdbcReader.getType(
+                            metaData.getColumnType(i + 1), metaData.getColumnTypeName(i + 1));
+        }
+
+        TableSchema tableSchema = tableSchemaMap.get(topicName);
+        if (tableSchema == null) {
+            tableSchema =
+                    OceanBaseTableSchema.getTableSchema(
+                            topicName, databaseName, tableName, columnNames, jdbcTypes);
+            tableSchemaMap.put(topicName, tableSchema);
+        }
+
+        Struct source = OceanBaseSourceInfo.struct(tenantName, databaseName, tableName, null, null);
+
+        while (rs.next()) {
+            Struct value = new Struct(tableSchema.valueSchema());
+            for (int i = 0; i < metaData.getColumnCount(); i++) {
+                value.put(
+                        columnNames[i],
+                        OceanBaseJdbcReader.getField(jdbcTypes[i], rs.getObject(i + 1)));
+            }
+            Struct struct = tableSchema.getEnvelopeSchema().create(value, source, null);
+            deserializer.deserialize(
+                    new SourceRecord(
+                            partition,
+                            offset,
+                            topicName,
+                            null,
+                            null,
+                            null,
+                            struct.schema(),
+                            struct),
+                    outputCollector);
+        }
+    }
+
+    protected void readChangeEvents() throws InterruptedException {
+        ObReaderConfig obReaderConfig = new ObReaderConfig();
+        obReaderConfig.setRsList(rsList);
+        obReaderConfig.setUsername(username);
+        obReaderConfig.setPassword(password);
+        obReaderConfig.setTableWhiteList(
+                String.format("%s.%s.%s", tenantName, databaseName, tableName));
+
```

Comment:
I think we can support read multiple table easily as `LogProxyClient` offer the ability here.